app/services/PaymentAPIService.scala

```diff
@@ -2,12 +2,28 @@ package services
 
 import java.io.IOException
 
+import io.circe.generic.JsonCodec
+import io.circe.parser.parse
+import codecs.CirceDecoders.decodeErrorWrapper
 import play.api.libs.json._
 import play.api.libs.ws.{WSClient, WSResponse}
 import services.ExecutePaymentBody._
+import monitoring.SafeLogger
 
 import scala.concurrent.{ExecutionContext, Future}
 
+class PaymentApiError extends Exception
```

Comment:
this can be a `sealed trait`; I don't think it's necessary for it to extend `Exception`